packages/dd-trace/src/appsec/api_security_sampler.js

```diff
@@ -1,61 +1,87 @@
 'use strict'
 
+const LRUCache = require('lru-cache')
+const web = require('../plugins/util/web')
 const log = require('../log')
+const { AUTO_REJECT, USER_REJECT } = require('../../../../ext/priority')
 
-let enabled
-let requestSampling
+const MAX_SIZE = 4096
+const DEFAULT_DELAY = 30 // 30s
 
-const sampledRequests = new WeakSet()
+let enabled
+let sampledRequests
 
 function configure ({ apiSecurity }) {
   enabled = apiSecurity.enabled
-  setRequestSampling(apiSecurity.requestSampling)
+  const delay = apiSecurity.sampleDelay || DEFAULT_DELAY
+  sampledRequests = new LRUCache({ max: MAX_SIZE, ttl: delay * 1000 })
 }
 
 function disable () {
   enabled = false
+  sampledRequests?.clear()
 }
 
-function setRequestSampling (sampling) {
-  requestSampling = parseRequestSampling(sampling)
-}
+function sampleRequest (req, res, force = false) {
+  if (!enabled) return false
+  if (this.isSampled(req, res)) return false
```

Comment:
why `this.`? Isn't this code will crash if:

```
const { sampleRequest } = require('..../api_security_sampler.js')
sampleRequest(req, res)
```


```suggestion
  if (isSampled(req, res)) return false
```